packages/expo-file-system/src/next/FileSystem.ts

```diff
@@ -0,0 +1,21 @@
+import ExpoFileSystem from './ExpoFileSystem';
+import { Path } from './FileSystem.types';
+
+export class File extends ExpoFileSystem.FileSystemFile {
+  constructor(path: Path) {
+    super(path);
+    this.validatePath();
+  }
+}
+
+export class Directory extends ExpoFileSystem.FileSystemDirectory {
+  constructor(path: Path) {
+    super(path);
+    this.validatePath();
+  }
+}
+
+// consider module functions as API alternative
+export async function write(file: File, contents: string) {
```

Comment:
I think we won't need this, especially with this name 😛 It seems quite weird that we're mixing functional approach with object-oriented (it takes a `File` as an input).

Reply:
Again, it's a strong bun inspiration, plus I think it's a good idea to keep the module level functions, maybe also accepting string as file and trying to create a file from that.
```
const input = Bun.file("input.txt");
const output = Bun.file("output.txt"); // doesn't exist yet!
await Bun.write(output, input);
```